### ISSUE TITLE
[core] : URL validation for new pages under Site root is not correct

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#19831]
Found when working on Support ticket: [103347](https://dotcms.zendesk.com/agent/tickets/103347)
Reproducible as of dotCMS 5.2.8, and latest master.

Validating the URL of an HTML Page created at the root of a Site is not being done correctly. The mechanism used to retrieve the Site that the HTML Page is being created in is not correct.

**NOTE: The existing process DOES return an error when a new HTML Page has the same URL as another page, file, or folder. However, that's not being validated in the right method. The error is being triggered by a subsequent process.**

**To Reproduce**
1. At the root of your Site, create an HTML Page called `my-test`
2. Then, try to create another page with the exact same URL.
3. You'll see the following error message:
```
Page URL [/index] already exists with content ID [a9f30020-54ef-494e-92ed-645e757171c2]
```
Even though the error message correctly describes the error situation, that is not the expected message.

**Expected behavior**
The correct error message in this situation should be the following:
```
Another Page with the same URL already exists in this folder
```

**Additional context**

**Comment by jcastro-dotcms** [comment]
PR: https://github.com/dotCMS/core/pull/19833

**Comment by dsilvam** [comment]
Passed Internal QA: proper message logged and displayed in the UI

**Comment by bryanboza** [comment]
Fixed, tested on release-21.04// Postgres // FF
